s3/src/main/resources/reference.conf

```diff
@@ -77,8 +77,13 @@ alpakka.s3 {
   # AWS S3 is going to retire path-style access, thus prefer the virutal-host-style access
   # https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/
   path-style-access = false
+  # Don't use! Possibility to switch off the logged warning when using `path-style-access`.
```

Comment:
A bit surprising that it starts with "Don't use!".

An alternative solution to introducing a new flag could be a new value to the old flag that signifies that the user has understood what they are doing. Like `path-style-access = force` or `path-style-access = yes-I-know-what-I-am-doing`.